### PR TITLE
feat: [CCM-11128]: Added CE settings in default settings page

### PR DESCRIPTION
**Pull request by jyotiarora-harness**
### Summary

CCM settings added in default settings page

#### Screenshots

Screenshot 2023-04-03 at 11 11 05 AM
Screenshot 2023-04-03 at 11 16 15 AM

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/23-default-settings/components/ReusableHandlers.tsx [hunk 1]
```diff
@@ -171,3 +171,23 @@ export const DefaultSettingDurationField: React.FC<SettingRendererProps> = ({
...
+  onSettingSelectionChange,
+  settingValue,
+  identifier,
+  errorMessage
+}) => {
+  return (
+    <Layout.Vertical flex={{ alignItems: 'flex-start' }} className={css.settingCheckBoxContainer}>
+      <Toggle
+        label=""
+        checked={settingValue?.value === 'true'}
```
Comment (swarajcheguri-harness) [inline src/modules/23-default-settings/components/ReusableHandlers.tsx:185]: is this always true as string or boolean value also should work ?

Comment (jyotiarora-harness) [reply src/modules/23-default-settings/components/ReusableHandlers.tsx:185]: In BE contract it's defined as string type

**Review by swarajcheguri-harness — APPROVED** [review]

File: src/modules/10-common/navigation/SideNav/SideNav.tsx [hunk 2]
```diff
@@ -158,11 +158,19 @@ interface SidebarLinkProps extends NavLinkProps {
...
+  label,
+  icon,
+  rightIcon,
+  className,
+  textProps,
+  ...others
+}) => (
   <Link className={cx(css.link, className)} activeClassName={css.selected} {...others}>
-    <Text icon={icon} className={css.text} {...textProps}>
+    <Text icon={icon} rightIcon={rightIcon} className={css.text} {...textProps}>
```
Comment (rajarshee-harness) [inline src/modules/10-common/navigation/SideNav/SideNav.tsx:173]: We could pass the `rightIcon` in `textProps`

File: src/modules/23-default-settings/RouteDestinations.tsx [hunk 3]
```diff
@@ -60,6 +63,135 @@ DefaultSettingsFactory.registerCategory('PMS', {
   modulesWhereCategoryWillBeDisplayed: ['cd', 'ci']
 })
 
+DefaultSettingsFactory.registerCategory('CE', {
+  icon: 'ccm-solid',
+  label: 'common.purpose.ce.continuous',
+  modulesWhereCategoryWillBeDisplayed: ['ce']
+})
+
+DefaultSettingsFactory.registerGroupHandler(SettingGroups.PERSPECTIVES_PREFERENCES, {
```
Comment (rajarshee-harness) [inline src/modules/23-default-settings/RouteDestinations.tsx:72]: Should we register these in `CESideNav`?

**Review by rajarshee-harness — APPROVED** [review]

File: src/modules/23-default-settings/RouteDestinations.tsx (hunk 3 above)
Comment (jyotiarora-harness) [reply src/modules/23-default-settings/RouteDestinations.tsx:72]: All the settings related things they are registering in settings Route only for better clarity

**Pull request merged**